File: Backend/src/modules/armory/material/character_info.rs
use crate::modules::armory::material::Gear;

#[derive(Debug)]
pub struct CharacterInfo {
  pub id: u32,
  pub gear: Gear,
  pub hero_class_id: u8,
  pub level: u8,
  pub gender: bool,
  pub profession1: Option<u8>,
  pub profession2: Option<u8>,
  pub talent_specification: Option<String>,
  pub faction: bool,
  pub race_id: u8
}

File: Webclient/src/app/module/armory/domain_value/character_info.ts
import {CharacterGear} from "./character_gear";

export interface CharacterInfo {
    id: number;
    gear: CharacterGear;
    hero_class_id: number;
    level: number;
    gender: boolean;
    profession1: number | null;
    profession2: number | null;
    talent_specialization: string | null;
    race_id: number;
}
